rir/src/compiler/translations/rir_2_pir/rir_2_pir.h

```diff
@@ -18,50 +18,27 @@ class Rir2Pir {
             const std::string& name)
         : compiler(cmp), srcFunction(srcFunction), log(log), name(name) {}
 
+    bool tryCompile(Builder& insert) __attribute__((warn_unused_result)) {
+        return tryCompile(srcFunction->body(), insert);
+    }
+
+  private:
     // Tries to compile the srcCode. Return value indicates failure. Builder
     // has to be discarded, if compilation fails!
     bool tryCompile(rir::Code* srcCode, Builder& insert)
         __attribute__((warn_unused_result)) {
-        return translate<bool>(srcCode, insert,
-                               [&](Value* v) {
-                                   finalize(v, insert);
-                                   return true;
-                               },
-                               []() { return false; });
+        if (auto res = tryTranslate(srcCode, insert)) {
+            finalize(res, insert);
```

Comment:
Finalize should not fail.

tryCompile fails = unsupported feature
finalize fails = bug

for example the compiler should always generate code such that insertcast succeeds. We do insert cast lazily, because sometimes it's hard to do it eager, as you do not know all the usages of a value when you create it